Sources/OfflineEntitlements/PurchasedProductsFetcher.swift

```diff
@@ -0,0 +1,122 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PurchasedProductsFetcher.swift
+//
+//  Created by Andrés Boedo on 3/17/23.
+
+import Foundation
+import StoreKit
+
+/// This struct should have all the information we need from StoreKit
+/// to create EntitlementInfo from a StoreKit 2 transaction.
+/// Other fields from other places might be needed.
+@available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+struct PurchasedSK2Product {
+
+    let productIdentifier: String
+    let periodType: PeriodType
+    let isActive: Bool
+    let willRenew: Bool
+    let latestPurchaseDate: Date?
+    let originalPurchaseDate: Date?
+    let expirationDate: Date?
+    let store = Store.appStore
+    let isSandbox: Bool
+    let unsubscribeDetectedAt: Date?
+    let billingIssueDetectedAt: Date?
+    let ownershipType: PurchaseOwnershipType
+    let verification: VerificationResult = .verified
+
+    init(from transaction: StoreKit.Transaction,
```

Comment:
Small code clarity suggestion that I like doing: separate the `struct` definition and fields from this `init`, putting it in an extension. So it's easier to see at a glance what the `struct` has when using it, versus implementation detail on how it's constructed.

Reply:
I like it! done